Taskfile.yml

```diff
@@ -0,0 +1,166 @@
+
+  backend-up:
+    desc: Start backend web server
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml up -d
+
+  backend-down:
+    desc: Stop backend web server
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml down
+
+  backend-logs:
+    desc: Show and follow backend web server logs
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml logs -f
+
+  backend-restart:
+    desc: Restart backend web server
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml restart -d
+
+  backend-bash:
+    desc: Open a shell in the backend container
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml exec web sh
+
+  backend-shell:
+    desc: Open a Django shell in the backend container
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml exec web sh -c "python ./manage.py shell"
+
+  backend-pytest:
+    desc: 'Run pytest in the backend container E.g: task backend-pytest PYTEST_ARGS="tdpservice/test/ -s -vv"'
+    dir: tdrs-backend
+    vars:
+      PYTEST_ARGS: '{{.PYTEST_ARGS | default "."}}'
+    cmds:
+      - task backend-up
+      - docker-compose -f docker-compose.yml exec web sh -c "pytest {{.PYTEST_ARGS}}"
+
+  backend-lint:
+    desc: Run flake8 in the backend container
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml exec web sh -c "flake8 . && if [ $? -eq 0 ]; then echo 'Flake8 linter found no issues'; fi"
+
+  clean:
+    desc: Remove all containers, networks, and volumes
+    cmds:
+      - docker-compose -f tdrs-backend/docker-compose.yml down -v
+      - docker-compose -f tdrs-frontend/docker-compose.yml down -v
+      - docker system prune -f -a
+  
+  clamav-up:
+    desc: Start clamav service
+    dir: tdrs-backend
+    cmds:
+      - docker-compose -f docker-compose.yml up -d clamav-rest
+
+  frontend-up:
+    desc: Start frontend web server
+    dir: tdrs-frontend
+    cmds:
+      - docker-compose -f docker-compose.yml up -d
+
+  frontend-down:
+    desc: Stop frontend web server
+    dir: tdrs-frontend
+    cmds:
+      - docker-compose -f docker-compose.yml down
+
+  frontend-restart:
+    desc: Restart frontend web server
+    dir: tdrs-frontend
+    cmds:
+      - docker-compose -f docker-compose.yml restart -d
+
+  frontend-av:
+    desc: Start frontend with optional clamav service
+    dir: tdrs-frontend
+    cmds:
+      - task: frontend-up
+      - task: clamav-up
+
+  
+  # need more work
+  frontend-init:
+    desc: Initialize the frontend project
+    dir: tdrs-frontend
+    cmds:
+      - docker-compose -f docker-compose.yml up -d --build
+      - docker-compose -f docker-compose.yml exec tdp-frontend sh -c "apk add nodejs npm"
+      - docker-compose -f docker-compose.yml exec tdp-frontend sh -c "npm install"
+      - docker-compose -f docker-compose.yml down
+
+  frontend-test:
```

Comment:
I cannot get it to exit when it's done, nor accepts input. have to use ctrl+c to exit.

Reply:
run in the background, and then exec terminal into the docker instance

Reply:
try: npm test instead of npm run test

Reply:
another idea is to : "test": "CI=true react-scripts test"